Lib/unittest/mock.py

```diff
@@ -1927,16 +1927,16 @@ def mock_add_spec(self, spec, spec_set=False):
 
 class MagicProxy(object):
     def __init__(self, name, parent):
-        self.name = name
-        self.parent = parent
+        self._mock_name = name
+        self._mock_parent = parent
 
     def __call__(self, *args, **kwargs):
         m = self.create_mock()
         return m(*args, **kwargs)
 
     def create_mock(self):
-        entry = self.name
-        parent = self.parent
+        entry = self._mock_name
+        parent = self._mock_parent
```

Comment:
Please can you add a unit test to show why this is needed?

Reply:
(Removed, see above)